src/evm/protocol/filters.rs

```diff
@@ -123,3 +123,22 @@ pub fn curve_pool_filter(component: &ComponentWithState) -> bool {
     }
     true
 }
+
+/// Filters out pool that have hooks in Uniswap V4
+pub fn uniswap_v4_pool_with_hook_filter(component: &ComponentWithState) -> bool {
+    if let Some(hooks) = component
+        .component
+        .static_attributes
+        .get("hooks")
+    {
+        if !hooks.to_vec().iter().all(|&b| b == 0) {
+            debug!("Filtering out UniswapV4 pool {} because it has hooks", component.component.id);
+            return false;
+        }
+    }
+    debug!(
+        "UniswapV4 pool with static attributes {:?} will not be filtered out.",
```

Comment:
I think I wanna change this, why do we do this in others filters?

Reply:
If you based this off the Balancer filter it was purely for my debugging because I couldn't figure out why some pools were not getting filtered out. Feel free to remove this!

Reply:
yeeess feel free to delete these logs! I had already deleted them but then got brought back from the dead in some bad merge